discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -455,10 +455,11 @@ def get_node_ips_and_macs(self, node_name):
 
     @staticmethod
     def _get_domain_ips_and_macs(domain):
-        # getting all DHCP leases IPs
-        interfaces = dict(domain.interfaceAddresses(libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE))
-        # getting static IPs via ARP
-        interfaces.update(domain.interfaceAddresses(libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_ARP))
+        interfaces = dict()
+
+        for addresses_source in (libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE, libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_ARP):
```

Comment:
copying comment from the other PR
I think we should leave the comments in some form, because it's not that immediate as for why we need those two methods
libvirt documentation is also not that great, so we should leave the original intent in the code imho